plugins/beacon/jvm/src/main/kotlin/com/intuit/playerui/plugins/beacon/BeaconPlugin.kt

```diff
@@ -68,6 +88,50 @@ public class BeaconPlugin(override val plugins: List<JSPluginWrapper>) : JSScrip
         )
     }
 
+    @Serializable(Hooks.Serializer::class)
+    public class Hooks internal constructor(override val node: Node) : NodeWrapper {
+        /** A hook to build beacon */
+        public val buildBeacon: NodeAsyncWaterfallHook2<Any?, HookArgs>
+            by NodeSerializableField(NodeAsyncWaterfallHook2.serializer(GenericSerializer(), HookArgs.serializer()))
+
+        /** A hook to cancel beacon */
+        public val cancelBeacon: NodeSyncBailHook1<HookArgs, Boolean>
+            by NodeSerializableField(NodeSyncBailHook1.serializer(HookArgs.serializer(), Boolean.serializer()))
+
+        internal object Serializer : NodeWrapperSerializer<Hooks>(::Hooks)
+    }
+
+    @Serializable(HookArgs.Serializer::class)
+    public class HookArgs internal constructor(override val node: Node) : NodeWrapper {
+        /** The current player state */
+        public val state: PlayerFlowState? by NodeSerializableField(PlayerFlowState.serializer().nullable)
+
+        /** The beacon plugin logger */
+        public val logger: LoggerType by NodeSerializableField(
+            MapSerializer(
+                LogSeverity.serializer(),
+                InvokableSerializer(GenericSerializer()) as KSerializer<Invokable<Unit>>,
+            ),
+        )
+
+        /** The action being performed */
+        public val action: String by NodeSerializableField(String.serializer())
+
+        /** The specific element that the beacon originated from */
+        public val element: String by NodeSerializableField(String.serializer())
+
+        /** The asset firing the beacon */
+        public val asset: Asset by NodeSerializableField(Asset.serializer())
+
+        /** The current view */
+        public val view: Asset? by NodeSerializableField(Asset.serializer().nullable)
```

Comment:
is this of the right type?

Reply:
I think so? I used `initialView` from `View.kt` as an example.